GH-8680: Check DB for table on start (#8690)
spring-projects/spring-integration: src/reference/asciidoc/jdbc.adoc

Note that the scripts are provided as samples and as specifications of the required table and column names.
You may find that you need to enhance them for production use (for, example, by adding index declarations).

> Starting with version 6.2, the `JdbcMessageStore`, `JdbcChannelMessageStore`, `JdbcMetadataStore`, and `DefaultLockRepository` implement `SmartLifecycle` and perform `SELECT COUNT` query in their `start()` method to be sure that required table (according to the provided prefix) is present in the target database.
> If tables don't exist, the application context fails to start.

Reviewer: ```suggestion
Starting with version 6.2, the `JdbcMessageStore`, `JdbcChannelMessageStore`, `JdbcMetadataStore`, and `DefaultLockRepository` implement `SmartLifecycle` and perform a`SELECT COUNT` query, on their respective tables, in the `start()` method to ensure that the required table (according to the provided prefix) is present in the target database.
If the required table does not exist, the application context fails to start.
```
